### ISSUE TITLE
H65: "aria-label" attribute  instead of "title" atribute

### ISSUE THREAD

**Issue by MakotoUeki** [w3c/wcag#2354]
We might need to encourage to use the aria-label attribiute instead of the "title" attribute when the label element cannot be used.

In the WAI's [Labeling Controls in Forms Tutorial](https://www.w3.org/WAI/tutorials/forms/labels/), it reads:
> **Approach 4: Using the title attribute**
> The title attribute can also be used to identify form controls. This approach is generally less reliable and not recommended because some screen readers and assistive technologies do not interpret the title attribute as a replacement for the label element, possibly because the title attribute is often used to provide non-essential information. ......

WCAG Techniques must be consistent with the WAI resources, and vice versa. If WCAG won't encourage to use the title attribute any more, then H65 Technique should be updated soon.

**Comment by detlevhfischer** [comment]
If `title` is consistently exposed in the absence of another scource of the accessible name, I see no disadvantage in using it instead of `aria-label`. Is there evidence that when lacking another source, it is **not** exposed in any common OS/UA/SR scenario? Don't forget the not insignificant advantage over `aria-label` that `title` produces a native tootip in desktop / mouse user scenarios that will be helpful for sighted users, especially when using screen magnification and therefore lacking the wider visual context of in input.

**Comment by mbgower** [comment]
> because some screen readers and assistive technologies do not interpret the title attribute as a replacement for the label element

If an AT does that, they are not following the computation for Accessible Name. 

I think H65 as a sufficient technique is fine. It's not the only technique, just sufficient. Like @detlevhfischer, I also feel like title, while having challenges to do with keyboard surfacing, represents real value for many users with low vision or cognitive disabilities.

@MakotoUeki, listing https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA6 or a similar modified aria-label technique as sufficient for [Labels or Instructions](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-cues.html) needs a bit of consideration, and some cautionary language in the text of the technique, IMO.

aria-label will override any other form of accessible name. In my experience, it is frequently misused/abused. It also is more easy to overlook in regard to localization. I'm not sure why it is listed ahead of aria-labelledby and title in that document. Personally, I'd use it as a last resort.

**Comment by patrickhlauke** [comment]
> If an AT does that, they are not following the computation for Accessible Name.

isn't it step "E" in the computation algorithm? https://w3c.github.io/accname/#computation-steps

**Comment by bruce-usab** [comment]
The first rule of aria is to use something else when you can.  Lots of elements support title, so I am not understanding the harm.

Probably OT:  Recently it came to my attention that if an element like IMG has both alt and title, calculation for Accessible Name uses alt and Accessible Description uses title.  I want to look into this more, since that seems backwards.

**Comment by mbgower** [comment]
> if an element like IMG has both alt and title, calculation for Accessible Name uses alt and Accessible Description uses title

Yes, that's my recollection. A specific ALT is higher in the hierarchy, so it makes sense.

**Comment by bruce-usab** [comment]
Thank you both.  I wrote <q>seems backwards</q> since Names are short and Descriptions long; TITLEs tend to be short and ALT tends to long.

It makes sense from the perspective of Accessible Name _being_ the thing while Accessible Description is _about_ the thing.  ALT _is_ the (text) equivalent.  Title is supplemental, not the equivalent, but is _about_ the thing, even if short.

**Comment by Omer6325** [comment]
> `title`Erişilebilir ismin başka bir kaynağının yokluğunda sürekli olarak ifşa edilirse, bunun yerine kullanılmasında herhangi bir dezavantaj görmüyorum `aria-label`. Başka bir kaynak olmadığında, herhangi bir yaygın OS/UA/SR senaryosunda açığa çıkmadığına dair **kanıt var mı? **Masaüstü / fare kullanıcı senaryolarında, özellikle ekran büyütme kullanırken ve bu nedenle girişin daha geniş görsel bağlamından yoksun olduğunda, gören kullanıcılar için yardımcı olacak yerel bir araç ipucu üretmenin önemsiz olmayan avantajını `aria-label`unutmayın .`title`

**Comment by mbgower** [comment]
@patrickhlauke not sure if that comment was deleted because you thought it was a random string, but thought I'd supply a  translation. Feel free to delete this comment if your reason was anything other than believing the comment was unintelligible:
> If it is constantly disclosed in the absence of another source of the accessible name, I don't see any disadvantage in using aria-label instead. Is there any evidence that it is not exposed in any common OS/UA/SR scenario in the absence of any other source? **Note the non-trivial advantage of aria-label of generating a native tooltip that will be helpful for sighted users in desktop/mouse user scenarios, especially when using screen magnification and therefore lacks the wider visual context of the input.

I realized @Omer6325's comment might point to a lack of context for this issue, so thought I'd put in a few things.

[H65: Using the title attribute to identify form controls when the label element cannot be used](https://www.w3.org/TR/WCAG20-TECHS/H65.html) is listed along with a bunch of other techniques at the bottom of situation A in 1.3.1 Info and Relationships, where "The technology provides semantic structure to make information and relationships conveyed through presentation programmatically determinable".  These last set of techniques are gathered under option #10:  "Making information and relationships conveyed through presentation programmatically determinable using the following techniques:"

As such, these are all techniques used where there is already information present in the UI. For that reason, a couple of the first techniques listed use aria-labelledby, for those situations where the text is in the UI. So, I want to first emphasize that this is a much preferred method to aria-label, since any changes in the UI due to translation or content updates will be immediately picked up. The value of aria-label would not be automatically updated. Further, since there is no means of exposing the value of the aria-label information without inspecting the code, it is less likely to be detected. Finally, if someone uses aria-label, it overrides _any_ other possible methods of naming the input, include default HTML methods.

There _are_ several aria-label sufficient techniques available in WCAG techniques (not listed under 1.3.1):

- [ARIA6: Using aria-label to provide labels for objects](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/ARIA6.html)
- [ARIA14: Using aria-label to provide an invisible label where a visible label cannot be used](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/ARIA14.html)

You'll notice that they come with warnings _and_ in the case of the ARIA14, are intended for situations where there _is_ no visible label. So, they definitely aren't as preferred in the context of this SC, and certainly are no better than the use of Title.

So, why include use of Title in such situations? One reason is that aria-label controls are completely invisible to non-AT users, whereas title values are displayed by anyone exploring a page with a mouse. So, for the significant number of mouse users who may need a bit of guidance, the tooltip that appears on hover provides crucial guidance. Almost as importantly, it is also more likely to be discovered by anyone carrying out a review of a page for translation, update, etc.

A secondary reason is that since both aria-label and title are in the accessible name computation, there is no disadvantage to using title, and since it is lower down in the preferred order, it also gracefully takes a back seat to other techniques (even becoming a potentially useful description). I hope my comments make it clear that aria-label has a number of disadvantages, which make it a poorer choice.

**Comment by patrickhlauke** [comment]
@mbgower that comment by Omer is simply Detlev's second comment here https://github.com/w3c/wcag/issues/2354#issuecomment-1117095878, run through translate...

**Comment by grahamharper** [comment]
@mbgower 

> I think H65 as a sufficient technique is fine.…
> … while having challenges to do with keyboard surfacing

but shouldn't the inability for keyboard-only and touch users to access the content of `title` make H65 insufficient? If not, how is it sufficient for these users?

Or is the scope of H65 only intended to be sufficient to identify form controls to screenreader users and users with a mouse and some other technique is required to make it fully accessible to keyboard-only/touch users?
